### DISCUSSION TITLE
Can google index a fully dynamic routes?

### DISCUSSION THREAD

**Discussion by vlad-gritsenko** [vercel/next.js#14565] [Help] [answered]
Hi 👋
My app has one static route- `'/'`, other routes are dynamic and based on categories/countries choosed by user, e.g. `'/skating-in-usa'` or `'/rafting-and-kayaking-in-brazil'`. Can google index such dynamic routes? Will it be possible to search for _'skating in usa'_ and see an exact page (`'/skating-in-usa'`) in google's search result list?

I'm new to Next.js and SEO and not sure how things work 100%
💖

**Comment by JonCognioDigital** [comment, marked as answer]
Yes, absolutely, that's the whole point of SSR. As long as you're using getServerProps and generating the page at runtime then it'll all get indexed properly. You can test this by typing your dynamic URLs into your browser and checking you get all the content in your page source.

If you're using getStaticProps and fallback:true to get it to generate all your dynamic pages statically then it's still possible but it gets very messy because it uses a javascript fallback while it's loading.

**Reply by VityaSchel** [reply]
I use fallback:blocking but google crawler says it cannot find referenced links. I implemented sitemap that is generated within api route which gets all my blog posts from database but didn't test it.

**Reply by steven-tey** [reply]
Quick update on @jon64digital's answer – we recently released an [update](https://github.com/vercel/next.js/discussions/28180) that changes the behavior of `fallback: true` to server-render content for browsers and only render loader skeletons for users, which makes it the perfect, SEO-optimized solution for building large-scale apps with dynamic routes and ISR! :)
